Rules/CommonScripts/KillMessages.as

```diff
@@ -288,11 +291,48 @@ void onPlayerDie(CRules@ this, CPlayer@ victim, CPlayer@ killer, u8 customdata)
 			CBlob@ victimblob = victim.getBlob();
 			CPlayer@ helper = getAssistPlayer(victim, killer);
 
-			bool kill = killerblob !is null && victimblob !is null && killerblob.isMyPlayer() && killerblob !is victimblob;
+			bool kill = killerblob !is null && victimblob !is null && killerblob !is victimblob;
 			bool assist = helper !is null && helper.isMyPlayer();
+
 			if (kill)
 			{
-				add_message(KillSpreeMessage(victim));
+				if (isServer())
+				{
+					if (killerblob.getTeamNum() != victimblob.getTeamNum())
+					{
+						killer.set_u32("kill time", getGameTime());
+						
+						if(!killer.exists("killstreak"))
+						{
+							killer.set_u8("killstreak", 1);
+						}
+						else
+						{
+							killer.add_u8("killstreak", 1);
+						}
+					}
+
+					if (victim.get_u8("killstreak") > 4)
+					{
+
+						uint16 victim_netid = victim.getNetworkID();
+						uint16 killer_netid = killer.getNetworkID();
+						uint16 kill_count = victim.get_u8("killstreak");
```

Comment:
`"killstreak"` is a u8 but you're saving it to a u16 variable and writing it to the CBitStream as a u16.

Reply:
This also occurs at line 372.

Reply:
You also need to fix the places where you read killstreak as a u16.

Reply:
You've fixed writing to the bitstream, but `kill_count` is still a uint16. Simple, probably pointless fix.